sdk/src/main/java/com/akkaserverless/javasdk/eventsourcedentity/EventSourcedEntityBase.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.eventsourcedentity;
+
+import com.akkaserverless.javasdk.Metadata;
+import com.akkaserverless.javasdk.Reply;
+import com.akkaserverless.javasdk.ServiceCall;
+import com.akkaserverless.javasdk.impl.reply.ErrorReplyImpl;
+import com.akkaserverless.javasdk.impl.reply.ForwardReplyImpl;
+import com.akkaserverless.javasdk.impl.reply.MessageReplyImpl;
+import com.akkaserverless.javasdk.impl.reply.NoReply;
+import com.akkaserverless.javasdk.reply.ErrorReply;
+import com.akkaserverless.javasdk.reply.ForwardReply;
+import com.akkaserverless.javasdk.reply.MessageReply;
+
+import java.util.function.Function;
+
+// FIXME rename to EventSourcedEntity when the old annotation is removed
+
+/** @param <S> The type of the state for this entity. */
+public abstract class EventSourcedEntityBase<S> {
+
+  /**
+   * Implement by returning the initial empty state object. This object will be passed into the
+   * command and event handlers, until a new state replaces it.
+   *
+   * <p>Also known as "zero state" or "neutral state".
+   *
+   * <p><code>null</code> is an allowed value.
+   */
+  protected abstract S emptyState();
```

Comment:
I wonder if we should pass the `entityId` to the `emptyState`?

Reply:
entityId will be passed to the factory lambda (constructor) via the CreationContext, so it's possible to keep that in an instance variable and accessed here